clar2wasm/src/words/sequences.rs

```diff
@@ -728,6 +738,14 @@ impl ComplexWord for Map {
             generator.visit_call_user_defined(&mut loop_, return_element_type, fname)?;
         }
 
+        smallest_sequence_length = if smallest_sequence_length == u32::MAX {
+            0
+        } else {
+            smallest_sequence_length
+        };
+
+        generator.frame_size += return_element_size as u32 * smallest_sequence_length;
```

Comment:
lines 634 -> 652 look like `create_call_stack_local` but with extra steps. 
Not sure how I feel about this:

- on the one hand, it seems this is "optimized" so we don't allocate more space than needed
- on the other hand, any change/optimization to `create_call_stack_local` won't be reflected here.

Reply:
I've a fix in mind, but another reason to avoid `create_call_stack_local` was we don't need to increase stack-pointer here.